src/modules/simulator/simulator_mavlink.cpp

```diff
@@ -100,89 +102,69 @@ mavlink_hil_actuator_controls_t Simulator::actuator_controls_from_outputs()
 
 		/* multirotors: set number of rotor outputs depending on type */
 
-		unsigned n;
-
 		switch (_system_type) {
 		case MAV_TYPE_AIRSHIP:
 		case MAV_TYPE_VTOL_DUOROTOR:
-			n = 2;
+			motors_count = 2;
 			break;
 
 		case MAV_TYPE_QUADROTOR:
 		case MAV_TYPE_VTOL_QUADROTOR:
 		case MAV_TYPE_VTOL_TILTROTOR:
-			n = 4;
+			motors_count = 4;
 			break;
 
 		case MAV_TYPE_VTOL_RESERVED2:
 			// this is the standard VTOL / quad plane with 5 propellers
-			n = 5;
+			motors_count = 5;
 			break;
 
 		case MAV_TYPE_HEXAROTOR:
-			n = 6;
+			motors_count = 6;
 			break;
 
+
 		default:
-			n = 8;
+			motors_count = 8;
 			break;
 		}
+	}
 
-		for (unsigned i = 0; i < 16; i++) {
-			if (armed) {
-				if (i < n) {
-					/* scale PWM out PWM_DEFAULT_MIN..PWM_DEFAULT_MAX us to 0..1 for rotors */
-					msg.controls[i] = (_actuator_outputs.output[i] - PWM_DEFAULT_MIN) / (PWM_DEFAULT_MAX - PWM_DEFAULT_MIN);
+	for (unsigned i = 0; i < actuator_outputs_s::NUM_ACTUATOR_OUTPUTS; i++) {
+		if (!armed) {
+			/* send 0 when disarmed and for disabled channels */
+			msg->controls[i] = 0.0f;
 
-				} else {
-					/* scale PWM out PWM_DEFAULT_MIN..PWM_DEFAULT_MAX us to -1..1 for other channels */
-					msg.controls[i] = (_actuator_outputs.output[i] - pwm_center) / ((PWM_DEFAULT_MAX - PWM_DEFAULT_MIN) / 2);
-				}
+		} else if ((motors_count == 0 && i == 4) ||			//Fixed wing, throttle
+			   (motors_count != 0 && i < motors_count)) {	//multirotor, roto channel
+			/* scale PWM out PWM_DEFAULT_MIN..PWM_DEFAULT_MAX us to 0..1 for rotors */
+			msg->controls[i] = (_actuator_outputs.output[i] - PWM_DEFAULT_MIN) / (PWM_DEFAULT_MAX - PWM_DEFAULT_MIN);
+			msg->controls[i] = math::constrain(msg->controls[i], 0.f, 1.f);
```

Comment:
I think this assumes that for fixedwing or unknown MAV_TYPE `motors_count=0`

However, I believe there are use cases that all actuators range fro `-1 ~ 1` and as far as I can see this logic would not be able to handle it.

Reply:
What are those cases?

Reply:
To name one, https://github.com/PX4/Firmware/pull/16004 What I was trying to say that fixedwing configuration shouldn't be the default

Reply:
@Jaeyoung-Lim I have added another commit, it should solve all cases